ncm-network/src/main/pan/components/network/core-schema.pan

```diff
@@ -28,11 +206,24 @@ type structure_interface_alias = {
     Describes the bonding options for configuring channel bonding on EL5 and similar.
 }
 type structure_bonding_options = {
-    "mode" : long(0..6)
+    "ad_select" ? string with match(SELF, '^(\d+|stable|bandwidth|count)$')
+    "all_slaves_active" ? long
+    "arp_interval" ? long
+    "arp_ip_target" ? type_ip[]
+    "arp_validate" ? string with match(SELF, '^(\d+|none|active|backup|all)$')
+    "downdelay" ? long
+    "fail_over_mac" ? string with match(SELF, '^(\d+|none|active|follow)$')
+    "min_links" ? long
+    "use_carrier" ? long
+    "num_grat_arp" ? long
+    "num_unsol_na" ? long
+    "resend_igmp" ? long
+    "mode" : long(0..6) # TODO: allow mode to be given as a string
```

Comment:
why?  dual typed values are not recommended. we have run into it on some other occasions, and i think they are used to migrate from legacy yes/no boolean strings to actual booleans.